app/views/comments/_form.html.erb

```diff
@@ -66,37 +65,64 @@
 
   <%= javascript_include_tag "dragdrop" %>
   <script>
-    $('#comment-form').bind('ajax:beforeSend', function(event){      
+    $('#comment-form').bind('ajax:beforeSend', function(event){
       $("#text-input").prop('disabled',true)
-      $("#comment-form .btn-primary").button('loading',true);      
+      $("#comment-form .btn-primary").button('loading',true);
     });
-    <% emoji_names = emoji_names_list %>
-    <% emojis = emoji_names[:emojis] %>
-    <% image_map = emoji_names[:image_map] %>
-    var image_map = <%= raw image_map.to_json %>
-    horsey(document.querySelector('textarea#text-input'), {
-        suggestions: (<%= raw emojis.to_json %>),
-        anchor: ':',
-        render: function (li, suggestion) {
-            var image = image_map[suggestion.text];
-            li.innerHTML = "<b>"+image+"</b><span>"+suggestion.text+"</span>";
+
+    var at_config = {
+      at: "@",
+      callbacks: {
+        remoteFilter: function(query, callback) {
+          $.getJSON("/api/srch/profiles?srchString=" + query, {}, function(data) {
+            if (data.hasOwnProperty('items') && data.items.length > 0) {
+              callback(data.items.map(function(i) { return i.docTitle }));
+            }
+           });
+          }
         },
         limit: 6
-    });
+      },
+      hashtags_config = {
+        at: "#",
+        callbacks: {
+          remoteFilter: function(query, callback) {
+            $.post('/tag/suggested/____' + query, function(response) {
+               callback(response.map(function(tagnames){ return tagnames }));
+             });
+            }
+          },
+        limit: 6
+      },
+      emojis_config = {
+        at: ':',
+        callbacks: {
+          remoteFilter: function(query, callback) {
+            $.getJSON("/emoji.json" + query, {}, function(data) {
```

Comment:
Ah, interesting, so here, we could remove the ` + query` part, because we're going to manually dig through the returned results to find what we want. 

You're noting that it returns 8 of them? I'm a little surprised it doesn't return the whole list, but maybe `atwho` is filtering it or something? I think we'd need to somehow filter this list for ones that match the query... not totally sure how to do that... looking at https://github.com/ichord/jquery-atwho-rails

Reply:
I think i see! I think it's returning the first 8. Once we remove ` + query`, we can optimize this in a follow-up!

Reply:
And we'll be able to refine what's displayed in the menu vs. inserted using these template settings! https://github.com/ichord/At.js/wiki/Base-Document#custom-template